### ISSUE TITLE
0.85: user object in the Extra formula

### ISSUE THREAD

**Issue by j2l** [saltcorn/saltcorn#1790]
There is now full access to the user object in the row inclusion formulae as of #1738 . This is done by partially evaluating the expression so that e.g. `{id: 5}.id` is reduced to `5` before sending it to the ORM query. Additions between constant terms are also resolved.

_Originally posted by @glutamate in https://github.com/saltcorn/saltcorn/issues/1638#issuecomment-1483419256_
            

I'd like to "filter" only the chat view the user is a participant of (only one per user), and since I don't get how to filter correctly to search into participants table, I added a `user.room` field that get populated when a participant is added to a room. 

Now, I expected the extra formula `{id:user.room}` to work but I get the error `Need room id`.

To recap:
users
- id
- room

+ traditional room setup (participants, ...) + room view

`url/view/room?id=1` works but when I embed the view into a page and extra state formula: `{id:user.room}` it doesn't get it and throws `Need room id`.

Any idea what I do wrong?
Or is there another better way to filter room based on user-room found on participants table?

**Comment by j2l** [comment]
I found a convoluted way:
A participants_show view exposes users and rooms tables. You can access join fields from there.
So it can embed the room view with the formula `{id:room.id}` based on the record shown.
Adding this participants_show view within a page with extra formula `{user:user.id}` works.
Basically, it gets the LoggedIn user.id, apply it to participants_show view.
In this participants_show record you get the room.id that you can use to filter the embedded view of the room `{id:room.id}`.

Is there an easier way?
